### DISCUSSION TITLE
accounts-js prisma adapter

### DISCUSSION THREAD

**Discussion by Stringsaeed** [accounts-js/accounts#1145] [Show and tell]
Here I created an accounts-js Prisma 2 adapter for all that has been asked for.

[prisma2-accounts-js-adapter](https://gist.github.com/Stringsaeed/589c4e9ce3c09ec46079d2549645866a) 

- ### First Step:
  Create `Databaseprovider.tsx` file, copy `DatabaseProvider.tsx` from gist
- ### Second Step: 
  initialize accountsjs server with the way you like but add to its config
  ```js
  const accountsServer = new AccountsServer(
  {
    db: new DatabaseProvider(),
    tokenSecret: 'SECRET',
  },
  { password },
  );
  ```
- ### Third Step
  copy `models.prisma` from Gist to your `schema.prisma`
  
  ---
  **NOTE**
  
  This because Prisma schema doesn't support extend or import 😔😔
  
  ---
---
**_Note: if you have any issue or bug feel free to comment here or comment to gist to update ASAP_**

**Comment by pradel** [comment]
Thanks for sharing this!

**Comment by CaptainN** [comment]
Thanks for this! It looks like maybe something got out of sync though. The typescript is telling me there are a few missing methods on the DatabaseProvider from DatabaseInterface.

```
Class 'DatabaseProvider' incorrectly implements interface 'DatabaseInterface<User>'.
  Type 'DatabaseProvider' is missing the following properties from type 'DatabaseInterface<User>': findUserByLoginToken, addLoginToken, removeAllLoginTokensts(2420)
```

I'm happy to contribute a patch, if someone can help me find the relevant docs. (I'll keep looking)

Update: It looks like they come from the `@accounts/magic-link` package. Neat!
